src/linear_algebra/exterior_algebra.lean

```diff
@@ -0,0 +1,145 @@
+
+variables (R : Type*) [comm_semiring R]
+variables (M : Type*) [add_comm_monoid M] [semimodule R M]
+
+namespace exterior_algebra
+open tensor_algebra
+
+/--
+An inductively defined relation on `tensor_algebra R M` used to define the exterior algebra.
+-/
+inductive rel : tensor_algebra R M → tensor_algebra R M → Prop
+| of (m : M) : rel ((ι R m) * (ι R m)) 0
+
+end exterior_algebra
+
+/--
+The exterior algebra of an `R`-semimodule `M`.
+-/
+@[derive [inhabited, semiring, algebra R]]
+def exterior_algebra := ring_quot (exterior_algebra.rel R M)
+
+namespace exterior_algebra
+
+variables {M}
+
+/--
+The canonical linear map `M →ₗ[R] exterior_algebra R M`.
+-/
+def ι : M →ₗ[R] exterior_algebra R M :=
+  (ring_quot.mk_alg_hom R _).to_linear_map.comp (tensor_algebra.ι R)
+
+
+variables {R}
+
+/-- As well as being linear, `ι m` squares to zero -/
+@[simp]
+theorem ι_square_zero (m : M) : (ι R m) * (ι R m) = 0 :=
+begin
+  dsimp [ι],
+  rw [←alg_hom.map_mul, ←alg_hom.map_zero _],
+  exact ring_quot.mk_alg_hom_rel R (rel.of m),
+end
+
+variables (R)
+
+/--
+Given a linear map `f : M →ₗ[R] A` into an `R`-algebra `A`, which satisfies the condition:
+`cond : ∀ m : M, f m * f m = 0`, this is the canonical lift of `f` to a morphism of `R`-algebras
+from `exterior_algebra R M` to `A`.
+-/
+def lift {A : Type*} [semiring A] [algebra R A] (f : M →ₗ[R] A) (cond : ∀ m, f m * f m = 0) :
+  exterior_algebra R M →ₐ[R] A :=
+ring_quot.lift_alg_hom R (tensor_algebra.lift R f)
+  (λ x y h, by {
+    induction h,
+    rw [alg_hom.map_zero, alg_hom.map_mul, tensor_algebra.lift_ι_apply, cond],
+  })
+
+@[simp]
+theorem ι_comp_lift {A : Type*} [semiring A] [algebra R A] (f : M →ₗ[R] A)
+(cond : ∀ m, f m * f m = 0) :
+  (lift R f cond).to_linear_map.comp (ι R) = f := by {ext, simp [lift, ι], }
+
+@[simp]
+theorem lift_ι_apply {A : Type*} [semiring A] [algebra R A] (f : M →ₗ[R] A)
+(cond : ∀ m, f m * f m = 0) (x) :
+  lift R f cond (ι R x) = f x := by { dsimp [lift, ι], rw tensor_algebra.lift_ι_apply, }
+
+@[simp]
+theorem lift_unique {A : Type*} [semiring A] [algebra R A] (f : M →ₗ[R] A)
+  (cond : ∀ m : M, f m * f m = 0)
+  (g : exterior_algebra R M →ₐ[R] A) : g.to_linear_map.comp (ι R) = f ↔ g = lift R f cond :=
+begin
+  refine ⟨λ hyp, _, λ hyp, by rw [hyp, ι_comp_lift]⟩,
+  ext,
+  -- TODO: why can't I combine these into `simp [lift, ← hyp]`?
+  simp [lift], rw ← hyp, simp,
+  refl,
+end
+
+attribute [irreducible] exterior_algebra ι lift
+
+variables {R M}
+
+@[simp]
+theorem comp_ι_square_zero {A: Type*} [semiring A] [algebra R A] (g : exterior_algebra R M →ₐ[R] A)
+  (m : M) : g (ι R m) * g (ι R m) = 0 :=
+by rw [←alg_hom.map_mul, ι_square_zero, alg_hom.map_zero]
+
+@[simp]
+theorem lift_comp_ι {A : Type*} [semiring A] [algebra R A] (g : exterior_algebra R M →ₐ[R] A) :
+  lift R (g.to_linear_map.comp (ι R)) (comp_ι_square_zero _) = g :=
+by {symmetry, rw ←lift_unique}
+
+theorem hom_ext {A : Type*} [semiring A] [algebra R A] {f g : exterior_algebra R M →ₐ[R] A} :
```

Comment:
I'd like to tag this as `@[ext]`, but it will result in goals of the form `⇑(f.to_linear_map.comp (ι R)) x = ⇑(g.to_linear_map.comp (ι R)) x`. Could you add an `@[ext]` lemma `f.to_linear_map (ι R x) = g.to_linear_map (ι R x) → f = g`?

Reply:
Good catch, this is already marked `ext` in
https://github.com/leanprover-community/mathlib/blob/da66bb8/src/algebra/free_algebra.lean#L271-L272

> Could you add an @[ext] lemma f.to_linear_map (ι R x) = g.to_linear_map (ι R x) → f = g?

Can I leave this to a follow up? This is also needed for `free_algebra` and `universal_enveloping_algebra`, so feels like it would be better to address for all of them at once.

Reply:
(I added the `ext` here)